### PR TITLE
glibc: 2.32-39 -> 2.32-40

### PR DESCRIPTION
**Pull request by TredwellGit**
###### Things done
- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
I recreated the archive and diffed them ``zdiff -u 2.32-master.patch.gz*``:

Is this expected?

```diff
--- /dev/fd/5	2021-03-28 08:05:54.136730618 +0200
+++ -	2021-03-28 08:05:54.145194553 +0200
@@ -255,20 +255,20 @@
 -      printf ("call 1 returned: %s\n", s);
 -      result = 1;
 -    }
--
--  bind_textdomain_codeset ("codeset", "UTF-8");
 +  TEST_COMPARE_STRING (gettext ("cheese"), "K\344se");
  
-   /* Here we expect output in UTF-8.  */
++  /* Here we expect output in UTF-8.  */
+   bind_textdomain_codeset ("codeset", "UTF-8");
++  TEST_COMPARE_STRING (gettext ("cheese"), "K\303\244se");
+ 
+-  /* Here we expect output in UTF-8.  */
 -  s = gettext ("cheese");
 -  if (strcmp (s, "K\303\244se"))
 -    {
 -      printf ("call 2 returned: %s\n", s);
 -      result = 1;
 -    }
-+  bind_textdomain_codeset ("codeset", "UTF-8");
-+  TEST_COMPARE_STRING (gettext ("cheese"), "K\303\244se");
- 
+-
 -  return result;
 +  /* `a with umlaut' is transliterated to `ae'.  */
 +  bind_textdomain_codeset ("codeset", "ASCII//TRANSLIT");
@@ -4243,6 +4243,11 @@
 -  _Unwind_Exception_Cleanup_Fn exception_cleanup;
 -  _Unwind_Word private_1;
 -  _Unwind_Word private_2;
+-
+-  /* @@@ The IA-64 ABI says that this structure must be double-word aligned.
+-     Taking that literally does not make much sense generically.  Instead we
+-     provide the maximum alignment required by any type for the machine.  */
+-} __attribute__((__aligned__));
 +  union
 +  {
 +    struct
@@ -4252,11 +4257,7 @@
 +      _Unwind_Word private_1;
 +      _Unwind_Word private_2;
 +    };
- 
--  /* @@@ The IA-64 ABI says that this structure must be double-word aligned.
--     Taking that literally does not make much sense generically.  Instead we
--     provide the maximum alignment required by any type for the machine.  */
--} __attribute__((__aligned__));
++
 +    /* The IA-64 ABI says that this structure must be double-word aligned.  */
 +    _Unwind_Word unwind_exception_align[2]
 +      __attribute__ ((__aligned__ (2 * sizeof (_Unwind_Word))));
```

**Comment by TredwellGit** [comment]
Different git diff algorithm. I will update to explicitly set it.
https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---diff-algorithmpatienceminimalhistogrammyers

File: pkgs/development/libraries/glibc/common.nix [hunk 1]
```diff
@@ -63,8 +63,8 @@ stdenv.mkDerivation ({
          and using git or something would complicate bootstrapping.
          Fortunately it's not too big.
           $ git checkout origin/release/2.32/master; git describe
-          glibc-2.32-39-g44b3959329
-          $ git show --reverse glibc-2.32.. | gzip -n -9 --rsyncable - > 2.32-master.patch.gz
+          glibc-2.32-40-g778b8d3786
+          $ git show --minimal --reverse glibc-2.32.. | gzip -n -9 --rsyncable - > 2.32-master.patch.gz
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/glibc/common.nix:67]:
```suggestion
          $ git show --minimal --reverse glibc-2.32.. | gzip -n -9 --rsyncable - > 2.32-master.patch.gz
          
          To compare the archive contents zdiff can be used.
          $ zdiff -u $file $file2
```

maybe a good information for reviewers.

**Review by SuperSandro2000 — APPROVED** [review]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 117858` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>stdenv</li>
</ul>
</details>

**Comment by Ma27** [comment]
Diff LGTM. Will take a look locally though :)

**Review by Ma27 — APPROVED** [review]
* Built a stdenv and a few small packages on `x86_64-linux` and `aarch64-linux`.
* Reviewed diff to old version of patchset-tarball

**Pull request merged**